COOKBOOK.md

```diff
@@ -0,0 +1,964 @@
+    <td align='center'>Custom Header and Footer</td>
+    <td align='center'>Custom Grid Layout</td>
+  </tr>
+</table>
+
+### Chat
+
+`Chat` is the next level down of context component from `OverlyProvider` that is required for `stream-chat-react-native` to function as designed. You can choose to wrap your entire application in `Chat` similar to what is required for the `OverlayProvider` or you can implement `Chat` at the screen level. `Chat` takes two important props, `client` and `i18nInstance`. The `client` should be an instance of StreamChat from [`stream-chat`](https://github.com/GetStream/stream-chat-js) configured for your app, and `i18nInstance` should be an instance of `Streami18n` from `stream-chat-react-native` configured for the desired language. `Chat` can also accept a `style` prop with the theme, this can be used to overwrite styles inherited from `OverlayProvider`. If you are using TypeScript you should add the appropriate generics to your instantiation of `StreamChat`, follow the documentation for [`stream-chat`](https://github.com/GetStream/stream-chat-js) to ensure proper setup.
+
+```tsx
+import { StreamChat } from 'stream-chat';
+import { Streami18n } from 'stream-chat-react-native';
+
+const streami18n = new Streami18n({ language: 'en' });
+const chatClient = StreamChat.getInstance<
+  AttachmentType,
+  ChannelType,
+  CommandType,
+  EventType,
+  MessageType,
+  ResponseType,
+  UserType
+>('key');
+```
+```typescript
+<Chat client={chatClient} i18nInstance={streami18n}>
+```
+
+### Channel
+
+When creating a chat screen it is required that `Channel` wrap the `stream-chat-react-native` components being used. `Channel` provides multiple contexts to the enclosed components and allows for modification of many of the enclosed components via props that are then kept in context.
+
+Three key props to `Channel` are `channel`, `keyboardVerticalOffset`, and `thread`. `channel` is a `StreamChat` channel. It can be created via `const channel = client.channel('type', 'id')` or is available as a callback on the `ChannelList` component via the prop `onSelect`. `keyboardVerticalOffset` is needed for adjusting the keyboard compatible view and should be the spacing above the `Channel` component, e.g. if you have a header it should be the header height. `thread` is a message object in the context of which a thread is occurring, i.e. the parent message of a thread. This can be set to any message, but is easily accessible on the `MessageList` component using the prop `onThreadSelect`. `Channel` also keeps an internal thread state which can be manipulated `openThread` and `closeThread` using the `ThreadContext` if you would prefer not to use your own state of the thread.
+
+```tsx
+const channel = client.channel('type', 'id');
+const headerHeight = useHeaderHeight();
+const { thread } = useContext(AppContext);
+```
+
+```tsx
+<Channel
+  channel={channel}
+  keyboardVerticalOffset={headerHeight}
+  thread={thread}
+>
+```
```

Comment:
I'll swap them to one, I was trying to imply code would be in between because I had started with the snippets being more "full" then that was getting to be too much boiler plate in the snippets and dropped the move.